In [2]:
#imports
from __future__ import division
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm
import pylab as pl
import numpy as np
%matplotlib inline
plt.style.use('ggplot')
import os
import csv
from sklearn import linear_model, cross_validation

# Project 4

In this project, you will summarize and present your analysis from Projects 1-3.

### Intro: Write a problem Statement/ Specific Aim for this project

Answer: Using UCLA admission dataset, we wanted to see if there was an association between admission and gre, gpa, and prestige of the students undergraduate school.

### Dataset:  Write up a description of your data and any cleaning that was completed

Answer: The dataset contained the variables 'admit', 'gre', 'gpa, and 'prestige'. We first needed to drop the "N/A" from the data and then we created dummy variables from the 'prestige' variable because it was a categorical variable not a continuous.

### Demo: Provide a table that explains the data by admission status

Mean (STD) or counts by admission status for each variable 

| Not Admitted | Admitted
---| ---|---
GPA | 3.347159(0.376355)  | 3.489206(0.371655)
GRE | 573.579336(116.052798) | 618.571429(109.257233)
Prestige 1 | 10.332 (%) | 26.1905 (%)
Prestige 2 | 35.06 (%) | 42.0635 (%)
Prestige 3 | 34.317 (%) | 22.2222 (%)
Prestige 4 | 20.295 (%) | 9.5238 (%)

In [22]:
df_raw = pd.read_csv("../assets/admissions.csv")
df_drop = df_raw.dropna()
df_new = df_drop.join(pd.get_dummies(df_raw.prestige, prefix = 'prestige'))
df_new.groupby('admit').describe()

gpa         gre    prestige  prestige_1.0  prestige_2.0  \
admit                                                                         
0     count  271.000000  271.000000  271.000000    271.000000    271.000000   
      mean     3.347159  573.579336    2.645756      0.103321      0.350554   
      std      0.376355  116.052798    0.918922      0.304941      0.478026   
      min      2.260000  220.000000    1.000000      0.000000      0.000000   
      25%      3.080000  500.000000    2.000000      0.000000      0.000000   
      50%      3.340000  580.000000    3.000000      0.000000      0.000000   
      75%      3.610000  660.000000    3.000000      0.000000      1.000000   
      max      4.000000  800.000000    4.000000      1.000000      1.000000   
1     count  126.000000  126.000000  126.000000    126.000000    126.000000   
      mean     3.489206  618.571429    2.150794      0.261905      0.420635   
      std      0.371655  109.257233    0.921455      0.441426      0.495632   
      min      2.420000  300.000000    1.000000      0.000000      0.000000   
      25%      3.220000  540.000000    1.000000      0.000000      0.000000   
      50%      3.545000  620.000000    2.000000      0.000000      0.000000   
      75%      3.757500  680.000000    3.000000      1.000000      1.000000   
      max      4.000000  800.000000    4.000000      1.000000      1.000000   

             prestige_3.0  prestige_4.0  
admit                                    
0     count    271.000000    271.000000  
      mean       0.343173      0.202952  
      std        0.475647      0.402941  
      min        0.000000      0.000000  
      25%        0.000000      0.000000  
      50%        0.000000      0.000000  
      75%        1.000000      0.000000  
      max        1.000000      1.000000  
1     count    126.000000    126.000000  
      mean       0.222222      0.095238  
      std        0.417399      0.294715  
      min        0.000000      0.000000  
      25%        0.000000      0.000000  
      50%        0.000000      0.000000  
      75%        0.000000      0.000000  
      max        1.000000      1.000000

### Methods: Write up the methods used in your analysis

Answer: After creating the dummy variables, we then split the data into our X and y variables that would be used in a logistic regression. Our y variable was 'admit' and we treated the rest as our X variables. Our model was able to give us a probabilty of admittance controlling for 'gpa' 'gre' and 'prestige'. We then were able to calculate the predicted probabilities for the students in our data.

### Results: Write up your results

Answer: Uding the model parameters and confidence intervals for those parameters: students who attended a tier 2 undergraduate school had 0.50 (CI 0.27, 0.94) the odds of being admitted to graduate school compared to students who attend a tier 1 undergraduate school. This decreased further with tier 3, OR=0.26(CI 0.13, 0.52), and tier 4, OR=0.21 (CI 0.09,  0.48), compared to tier 1 schools.  

### Visuals: Provide a table or visualization of these results

<img src='../assets/images/figure1.jpg' height= 75% width= 75%>

<img src='../assets/images/figure2.jpg' height= 75% width= 75%>

### Discussion: Write up your discussion and future steps

Answer: 

Our findings indicate that students have a higher probability of being admitted into grad school if they attended a prestigous undergraduate college by a statistically significant margin. Some next steps could include exploring differences by field of study or subdividing by Division sports team rankings. 